File: src/main/java/io/qross/sql/ForLoopVariables.java
package io.qross.sql;

import io.qross.util.DataCell;
import io.qross.util.DataRow;

import java.util.ArrayList;
import java.util.List;

public class ForLoopVariables {

    //循环列表项及值
    private List<DataRow> variables = new ArrayList<>();
    private int cursor = -1;

    public void addRow(DataRow row) {
        this.variables.add(row);
    }

    public boolean hasNext() {
        this.cursor++;
        return this.cursor < variables.size();
    }

    public boolean contains(String field) {
        if (this.variables.size() > 0) {
            return this.variables.get(0).contains(field);
        }
        else {
            return false;
        }
    }

    public DataCell get(String field) {
        return this.variables.get(this.cursor).getCell(field);
    }

    public void set(String field, Object value) {
        this.variables.get(this.cursor).set(field, value);
    }
}

File: src/main/scala/io/qross/psql/ForLoopVariables.scala
package io.qross.psql

import io.qross.core.{DataCell, DataRow}

import scala.collection.mutable.ArrayBuffer

class ForLoopVariables {
    //循环列表项及值
    private val variables = new ArrayBuffer[DataRow]
    private var cursor = -1

    def addRow(row: DataRow): Unit = {
        this.variables += row
    }

    def hasNext: Boolean = {
        this.cursor += 1
        this.cursor < variables.size
    }

    def contains(field: String): Boolean = {
        if (this.variables.nonEmpty) {
            this.variables.head.contains(field)
        }
        else {
            false
        }
    }

    def get(field: String): DataCell = this.variables(this.cursor).getCell(field)

    def set(field: String, value: Any): Unit = {
        this.variables(this.cursor).set(field, value)
    }
}